Tallies/TallyMaps/Maps1D/directionMap_class.f90

```diff
@@ -0,0 +1,194 @@
+module directionMap_class
+
+  use numPrecision
+  use universalVariables, only : valueOutsideArray, X_AXIS, Y_AXIS, Z_AXIS
+  use genericProcedures,  only : fatalError, dotProduct, numToChar
+  use dictionary_class,   only : dictionary
+  use grid_class,         only : grid
+  use particle_class,     only : particleState
+  use outputFile_class,   only : outputFile
+  use tallyMap1D_inter,   only : tallyMap1D, kill_super => kill
+
+  implicit none
+  private
+
+  !!
+  !! Maps the particle direction in linear bins in the range 0-360 degrees
+  !!
+  !! Interface:
+  !!   tallyMap1D interface
+  !!
+  !! NOTE:
+  !!   Behaviour of points exactly at the boundary of bins is undefined.
+  !!   particle can end-up in either of the two
+  !!
+  !! Sample Dictionary Input:
+  !!  directionMap {
+  !!    type directionMap;
+  !!    #plane xz;#            // Optional. Default xy
```

Comment:
For consistency with, say, cylinders and plotting, wouldn't it be preferable to have this as the perpendicular axis? E.g., xy would be z.

Reply:
I don't know because in a cylinder the perpendicular axis has more of a geometrical meaning than here. Here you're just literally scoping the angle in a 2D plane so I think the plane has more meaning

Reply:
Fair enough.